File: Advent of Code/2024/9.1.py
s = input()

a = []
id = 0
for i in range(len(s)):
    if i % 2:
        a.extend([-1] * int(s[i]))
    else:
        a.extend([id] * int(s[i]))
        id += 1

l, r = 0, len(a) - 1
while l <= r:
    if a[l] == -1:
        a[l], a[r] = a[r], a[l]
        while a[r] == -1:
            r -= 1
    l += 1

res = 0
for i in range(len(a)):
    if a[i] != -1:
        res += a[i] * i

print(res)

File: CodeRun/91.cpp
#include <iostream>
#include <algorithm>
#include <vector>
#include <set>
#include <unordered_map>
#include <string>
using namespace std;

int main() {
    freopen("input.txt", "r", stdin);
    freopen("output.txt", "w", stdout);

    int size, targetSum;
    cin >> size >> targetSum;

    unordered_map<int, int> seen;
    int sum = 0;
    seen[sum] = 1;

    long long res = 0;

    for (int i = 0; i < size; i++) {
        int value;
        cin >> value;

        sum += value;
        if (seen.count(sum - targetSum))
            res += seen[sum - targetSum];
        seen[sum]++;
    }

    cout << res;
}